ngraph/core/reference/include/ngraph/runtime/reference/region_yolo.hpp

```diff
@@ -0,0 +1,164 @@
+                                          int location,
+                                          int entry)
+            {
+                int n = location / (width * height);
+                int loc = location % (width * height);
+                return batch * outputs + n * width * height * (coords + classes + 1) +
+                       entry * width * height + loc;
+            }
+
+            static inline float logistic_activate(float x) { return 1.f / (1.f + std::exp(-x)); }
+            template <typename T>
+            static inline void
+                softmax_generic(const T* src_data, T* dst_data, int B, int C, int H, int W)
+            {
+                int start = 0;
+                for (unsigned int b = 0; b < B; b++)
+                {
+                    for (unsigned int i = start; i < H * W; i++)
+                    {
+                        float max = src_data[b * C * H * W + i];
+                        for (unsigned int c = 0; c < C; c++)
+                        {
+                            float val = src_data[b * C * H * W + c * H * W + i];
+                            if (val > max)
+                            {
+                                max = val;
+                            }
+                        }
+
+                        float sum = 0;
+                        for (unsigned int c = 0; c < C; c++)
+                        {
+                            dst_data[b * C * H * W + c * H * W + i] =
+                                std::exp(src_data[b * C * H * W + c * H * W + i] - max);
+                            sum += dst_data[b * C * H * W + c * H * W + i];
+                        }
+
+                        for (unsigned int c = 0; c < C; c++)
+                        {
+                            dst_data[b * C * H * W + c * H * W + i] /= sum;
+                        }
+                    }
+                }
+            }
+
+            template <typename T>
+            void region_yolo(const T* input,
+                             T* output,
+                             const Shape& input_shape,
+                             const int coords,
+                             const int classes,
+                             const int regions,
+                             const bool do_softmax,
+                             const std::vector<int64_t>& mask,
+                             const int axis,
+                             const int end_axis)
+            {
+                NGRAPH_CHECK(input_shape.size() == 4);
+
+                const int B = input_shape[0];
+                const int C = input_shape[1];
+                const int H = input_shape[2];
+                const int W = input_shape[3];
+
+                const auto mask_size = mask.size();
+                const int tensor_size = B * C * H * W;
+                for (unsigned int i = 0; i < tensor_size; i++)
+                {
+                    output[i] = input[i];
```

Comment:
I'd rather use std::copy instead of this raw loop to copy the full tensor.

Reply:
Agree. Changed it and combined with the previous comment